src/open_inwoner/accounts/models.py

```diff
@@ -678,23 +678,21 @@ def require_necessary_fields(self) -> bool:
         return False
 
     def get_logout_url(self) -> str:
-        # Exit early, because for some reason reverse("logout") fails after checking
-        # the singletonmodels
-        if not (self.is_bsn_user or self.is_eherkenning_user or self.is_eidas_user):
-            return reverse("logout")
-
-        if self.login_type == LoginTypeChoices.digid:
-            if OpenIDDigiDConfig.get_solo().enabled:
-                return reverse("digid_oidc:logout")
-            return reverse("digid:logout")
-        elif self.login_type == LoginTypeChoices.eherkenning:
-            if OpenIDEHerkenningConfig.get_solo().enabled:
+        match self.login_type:
+            case LoginTypeChoices.digid:
+                return (
+                    reverse("digid_oidc:logout")
+                    if OpenIDDigiDConfig.get_solo().enabled
+                    else reverse("digid:logout")
+                )
+            case LoginTypeChoices.eherkenning:
                 return reverse("eherkenning_oidc:logout")
-            return reverse("logout")
-        elif self.is_eidas_user:
-            if OpenIDEIDASConfig.get_solo().enabled:
+            case LoginTypeChoices.oidc:
+                return reverse("oidc_logout")
+            case _ if self.is_eidas_user:
```

Comment:
```suggestion
            case LoginTypeChoices.eidas_person_bsn | LoginTypeChoices.eidas_person_pseudo_id | LoginTypeChoices.eidas_company
```

a bit unusual to have two catch-all cases, one with a guard clause.

Reply:
Good point, and generally we should aim for exhaustiveness checking when using enums/textchoices. Your change makes that clearer, I added an `assert_never` as the fallthrough so each case has to be explicitly handled.